src/uct/ib/rc/base/rc_iface.c

```diff
@@ -239,7 +239,7 @@ ucs_status_t uct_rc_iface_query(uct_rc_iface_t *iface,
     iface_attr->cap.flags        |= UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
 
     /* Tag Offload */
-    uct_rc_iface_tag_query(iface, iface_attr, max_inline);
+    uct_rc_iface_tag_query(iface, iface_attr, max_inline, am_max_iov);
```

Comment:
i think eager iov should be larger than am iov since for am we also reserve some space for the user header which we don't have in case of eager, in case of eager we have TMH which should be calculated differently

Reply:
but the inline size of TMH is smaller than the am header size, no?

Reply:
Yes, but for IOVs we use the following macro:
`#define UCT_IB_MLX5_AM_ZCOPY_MAX_IOV  3UL`

Reply:
for eager_zcopy, we could have more IOVs